Sitka-Sound-Science-Center/sitka-landslide
# Archive rainfall.json with each site update

_UPDATE/note for future viewers of this PR: per [the comment below](https://github.com/azavea/sitka-landslide/pull/90#issuecomment-2407520285), commit [6cefc59](https://github.com/azavea/sitka-landslide/commit/6cefc597155aee17ced1f89358f1b3ff71e7d22e) was added as a follow-up to this, without a separate PR, adding the year to the S3 path._

## Overview

Adds an S3 bucket and modifies the build command to copy the `rainfall.json` file into it (with a timestamp added to the filename) every time the site builds.

Resolves #88

### Demo

Because the deployment setup for this project is quite simple and manual, and because the risk from making a mistake while running it is quite low (if anything goes wrong it will just fail to update the site, not actually break it), I went ahead and applied these changes.  So the bucket is already created and the build job has already started adding files to it.

![image](https://github.com/user-attachments/assets/9d140f77-28e7-43b3-83f4-1ecf5c7f404d)

![image](https://github.com/user-attachments/assets/dd4458f1-df88-4829-a2da-e31850a5fe56)

### Notes

This PR also includes some cleanup and small improvement items:
- Removes some `id_logit_3hr` files that have been sitting in the `data/` directory not doing anything (they're not referenced anywhere) and being confusing.
- Updates `site.tf` to swap a deprecated reference for the correct one and remove the geographic restrictions from the CloudFront distribution, which had been removed by hand in the console quite a while ago but were still there in the Terraform config.
- Updates the `cipublish` script to tag the published image with the git SHA in addition to `latest`.  The service will still just use the one with the `latest` tag, but we'll be able to see which container is which in list on ECR now.
- Moves the command that the Docker container runs to build the site into a script.  It was just inline in the Dockerfile, but the new command to copy to archive took it from "a bit complex" to "too much to just do inline here".  So now there's a `scripts/publish` within the `frontend/` directory and the container runs that.

## Testing Instructions

Since, as noted above, this is already deployed and working, I think just reading it over for errors or omissions is probably good enough.  You could also confirm that
- running `terraform plan` (from `deployment/terraform` and after initializing as described in the README) shows no changes
- running `docker build .` from `frontend/` successfully builds the container

## Checklist
